rotkehlchen/rotkehlchen.py

```diff
@@ -725,74 +726,69 @@ def query_balances(
 
-        stats: Dict[str, Any] = {
-            'location': {
-            },
-            'net_usd': net_usd,
-        }
-        for entry in total_usd_per_location:
-            name = entry[0]
-            total = entry[1]
-            if name == 'blockchain':  # blockchain is the only location with liabilities atm
-                total -= liabilities_total_usd
-
-            if net_usd != ZERO:
-                percentage = (total / net_usd).to_percentage()
-            else:
-                percentage = '0%'
-            stats['location'][name] = {
-                'usd_value': total,
+        # Calculate location stats
+        location_stats: Dict[str, Any] = {}
+        for location, total_usd in total_usd_per_location.items():
+            if location == str(Location.BLOCKCHAIN):
+                total_usd -= liabilities_total_usd
+
+            percentage = (total_usd / net_usd).to_percentage() if net_usd != ZERO else '0%'
+            location_stats[location] = {
+                'usd_value': total_usd,
                 'percentage_of_net_value': percentage,
             }
 
-        for k, v in combined.items():
-            if net_usd != ZERO:
-                percentage = (v['usd_value'] / net_usd).to_percentage()
-            else:
-                percentage = '0%'
-            combined[k]['percentage_of_net_value'] = percentage
-
-        for k, v in liabilities.items():
-            if net_usd != ZERO:
-                percentage = (v['usd_value'] / net_usd).to_percentage()
-            else:
-                percentage = '0%'
-            liabilities[k]['percentage_of_net_value'] = percentage
-
-        balance_sheet = {
-            'assets': combined,
-            'liabilities': liabilities,
+        # Calculate 'percentage_of_net_value' per asset
+        asset_total_balance_as_dict: Dict[Asset, Dict[str, Any]] = {
+            asset: balance.to_dict() for asset, balance in asset_total_balance.items()
+        }
+        liabilities_as_dict: Dict[Asset, Dict[str, Any]] = {
+            asset: balance.to_dict() for asset, balance in liabilities.items()
+        }
+        for asset, balance_dict in asset_total_balance_as_dict.items():
+            percentage = (balance_dict['usd_value'] / net_usd).to_percentage() if net_usd != ZERO else '0%'  # noqa: E501
+            asset_total_balance_as_dict[asset]['percentage_of_net_value'] = percentage
+
+        for asset, balance_dict in liabilities_as_dict.items():
+            percentage = (balance_dict['usd_value'] / net_usd).to_percentage() if net_usd != ZERO else '0%'  # noqa: E501
+            liabilities_as_dict[asset]['percentage_of_net_value'] = percentage
+
+        # Compose balances response
+        result_dict = {
+            'assets': asset_total_balance_as_dict,
+            'liabilities': liabilities_as_dict,
+            'location': location_stats,
+            'net_usd': net_usd,
         }
-
-        result_dict = merge_dicts(balance_sheet, stats)
```

Comment:
the `merge_dicts()` function is used nowhere else, so you can actually remove it.